### PR TITLE
Fix/cb 138

### PR DESCRIPTION
**Pull request by Kamikozz**
* fix: after changing wallet/userWalletAddress needs to re-fetch MyContracts data

* feat: clear all contracts forms when disconnecting wallet

* refactor:
  - walletService, add types for MapToDispatch etc.;
  - add `accountsChanged` to interactively change site's userWalletAddress;
  - disable running useEffect for connect in App component

### CODE REVIEW

File: src/store/myContracts/sagas/getMyContracts.ts [hunk 1]
```diff
@@ -200,18 +200,23 @@ function* fetchAndTransformContractsSaga(provider: Web3) {
...
+    const { data }: AxiosResponse<IGetContractsReturnType> = yield call(baseApi.getContracts, {
+      walletAddress: userWalletAddress,
+    });
+    const transformedData = yield call(
+      transformMyContractsDataSaga,
+      provider,
+      data,
+    );
+    const newContracts = createContractCards(transformedData);
+    console.log('New Conctracts Cards', newContracts);
```
Comment (ValnetineKul) [inline src/store/myContracts/sagas/getMyContracts.ts:214]: delete

File: src/store/contractForms/sagas/clearAllContractForms.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+} from 'redux-saga/effects';
+import {
+  deleteTokenContractForm,
+  deleteCrowdsaleContractForm,
+  deleteLostKeyContractForm,
+  deleteWillContractForm,
+  deleteWeddingContractForm,
+} from '../reducer';
+
+export function* clearAllContractFormsSaga() {
```
Comment (ValnetineKul) [inline src/store/contractForms/sagas/clearAllContractForms.ts:13]: you dont need saga for synchronous actions, just create clear all forms method in reducer and call it when you need to clear forms state

**Review by ValnetineKul — CHANGES_REQUESTED** [review]

File: src/store/contractForms/sagas/clearAllContractForms.ts (hunk 2 above)
Comment (Kamikozz) [reply src/store/contractForms/sagas/clearAllContractForms.ts:13]: I've just written this action in reducer
```ts
clearAllForms: (state) => ({
      ...state,
      ...initialState,
    }),
```
but, how to run this action on another reducer's action? like that:
```ts
export default function* listener() {
  yield takeLatest('user/disconnectWalletState', clearAllContractFormsSaga);
}
```
`extraReducers` field?

Comment (Kamikozz) [reply src/store/contractForms/sagas/clearAllContractForms.ts]: 1) removed saga
2) moved all clear logic into ConnectDropdownModal
```ts
  const disconnect = useCallback(async () => {
    onClose();
    dispatch(disconnectWalletState());
    dispatch(clearAllForms());
  }, [dispatch, onClose]);
  ```

**Pull request merged**